src/components/AddressDetailExplorer.js

```diff
@@ -0,0 +1,528 @@
+        });
+      });
+      return txhistory;
+    }).finally(() => {
+      this.setState({ loadingHistory: false });
+    });
+  }
+
+  reloadData = () => {
+    this.setState({
+      loadingTokens: true,
+    }, () => {
+      addressApi.getTokens(this.state.address, TOKEN_COUNT).then(response => {
+        if (!response) {
+          // An error happened with the API call
+          this.setState({ showReloadDataButton: true });
+          return;
+        }
+
+        let selectedToken = '';
+
+        if (!response) {
+          throw new Error('Error fetching tokens');
+        }
+
+        const tokens = response.tokens || {};
+        const total = response.total || 0;
+
+        if (total > Object.keys(tokens).length) {
+          // There were unfetched tokens
+          this.setState({ warnMissingTokens: total });
+        } else {
+          // This will turn off the missing tokens alert
+          this.setState({ warnMissingTokens: 0 });
+        }
+
+        if (this.state.selectedToken && tokens[this.state.selectedToken]) {
+          // use has a selected token, we will keep the selected token
+          selectedToken = this.state.selectedToken;
+        } else {
+          const hathorUID = hathorLib.constants.HATHOR_TOKEN_CONFIG.uid
+          if (tokens[hathorUID]) {
+            // If HTR is in the token list of this address, it's the default selection
+            selectedToken = hathorUID;
+          } else {
+            // Otherwise we get the first element, if there is one
+            const keys = Object.keys(tokens);
+            if (keys.length === 0) {
+              // In case the length is 0, we have no transactions for this address
+              this.setState({ loadingTokens: false });
+              return;
+            }
+            selectedToken = keys[0];
+          }
+        }
+
+        const tokenDidChange = selectedToken !== this.state.selectedToken || this.state.selectedToken === '';
+
+        this.setState({
+          addressTokens: tokens,
+          loadingTokens: false,
+          selectedToken,
+        }, () => {
+          if (tokenDidChange || !this.state.metadataLoaded) {
+            this.getSelectedTokenMetadata(selectedToken);
+          }
+
+          // Update token in the URL
+          this.updateTokenURL(selectedToken);
+          this.reloadTokenData(selectedToken);
+        });
+
+      }).catch(error => {
+        this.setState({
+          loadingTokens: false,
+          errorMessage: error.toString(),
+        });
+      });
+    });
+  }
+
+  reloadTokenData = (token) => {
+    this.setState({
+      loadingSummary: true,
+      loadingHistory: true,
+    }, () => {
+      addressApi.getBalance(this.state.address, token).then(response => {
+        if (!response) {
+          // An error happened with the API call
+          this.setState({ showReloadTokenButton: true });
+          return;
+        }
+        const balance = response || {};
```

Comment:
This was used before I had the if above to prevent api errors.
Our axios instance has an error handler which prevents errors from being thrown, so the `undefined` response is passed in this cases.
Since we already handle this i'll remove it.